### ISSUE TITLE
Split NZVH data resource (dr2153) into individual data resources for each institution

### ISSUE THREAD

**Issue by nielsklazenga** [AtlasOfLivingAustralia/data-management#1002]
**Problem**
- AK and WELT want to deliver data directly to ALA but cannot do so because their data is already in the NZVH data resource
- AK and WELT are not able to update their data in NZVH
- ALA has not been able to upload the NZVH data set because of duplicate row keys.

**Solution**
- Split the NZVH data set into individual data sets for each herbarium
- Herbarium that has duplicate row keys in its data will have to do a new data delivery.

**Comment by peggynewman** [comment]
Dependent on https://github.com/AtlasOfLivingAustralia/preingestion/issues/228

**Comment by peggynewman** [comment]
Waiting on devops to allow access to EMR studio to manipulate UUID AVRO

**Comment by rosemaryjoconnor** [comment]
**12/07/2024**
Associated Freshdesk ticket:   https://support.ehelp.edu.au/a/tickets/189866

**Comment by rosemaryjoconnor** [comment]
**24/06/2024**

**Current status**

**Databox - Avros split and data updated**
- dr2153 - NZ Virtual Herbarium (790,126)
- dr22717 - WELT (252,205)

**Production**

- dr2153 - Allan Herbarium (1,005,799)
- dr22717 - WELT (250,501)

**Comment by rosemaryjoconnor** [comment]
** 18/07/2024**

1. **Allan Herbarium - dr2153 now loading in production with IPT data
2. WELT Herbarium - dr26642 now loading in production with IPT data**

**WELT Institution and Collection codes not mapping**

- dr26642- Collection Code 'WELT' Institution Code: 'NMNZ'
- Provider Map update required

**Tasks**
- [ ] Create new Institution Code 'NMNZ' for https://collections.ala.org.au/institution/show/in82
- [ ] Update Provider Map: https://collections.ala.org.au/providerMap/show/139 with new Institution Code
- [ ] Re-load dr2153 Prod

**Comment by rosemaryjoconnor** [comment]
**18/07/2024**

Original dr2153 had multiple collections associated. Each of these needs Institution Code and Provider Map

- Records then need to be split from dr2153 and new Data resources created.
- Source of future updates unknown at this stage

**Collection Summary**

| ﻿CollectionCode | CollectionID | Collection Name | Recs in dr2153 | Institution | Inst ID | Institution Code | Provider Map |
| ---------------|--------------|-----------------|----------------|-------------|---------|------------------|--------------|
| AK | co249 | Auckland War Mem. Mus. Herbarium | 302043 | Auckland War Memorial Museum | in113 | ? | No |
| PDD | co215 | New Zealand Fungarium Te Kohinga Hekaheka o Aotearoa | 105756 | Landcare Research NZ Ltd | in92 | ? | No |
| NZFRI | co255 | National Forestry Herbarium | 28934 | Scion  | in119 | ? | No |
| CANU | co250 | Univ of Canterbury Herbarium | 17599 | University of Canterbury | in114 | ? | No |
| MPN | co254 | Dame Ella Campbell Herbarium | 16374 | Massey University | in118 | ? | No |
| LINC | co253 | Lincoln University Herbarium | 9295 | Lincoln University | in115 | ? | No |
| UNITEC | co218 | Unitec Inst. of Tech. Herbarium | 6752 | Unitec Institute of Technology | in99 | ? | No |
| NaN |  |  | 43 |  |  |  |  |

**Comment by rosemaryjoconnor** [comment]
**18/07/2024**
Next steps:
- [ ] Check for DRs for collections in Test/Prod
- [ ] Determine future update source
- [ ] Databox - Create/update provider codes/provider maps
- [ ] Prod- Create/update up provider codes/provider maps

**Comment by rosemaryjoconnor** [comment]
**18/07/2024**

Dataresources created in Test

**Dataresources - in test to date**

| ﻿CollectionCode | CollectionID | Collection Name |  DR | In GBIF | GBIF Link |
| ---------------|--------------|-----------------|----------------|---|---------|
| AK | co249 | Auckland War Mem. Mus. Herbarium |  dr22714 | Yes |  https://www.gbif.org/dataset/83ae84cf-88e4-4b5c-80b2-271a15a3e0fc |
| PDD | co215 | New Zealand Fungarium Te Kohinga Hekaheka.. |   |   |   |
| NZFRI | co255 | National Forestry Herbarium |   |  |  |
| CANU | co250 | Univ of Canterbury Herbarium |   |  |  | |
| MPN | co254 | Dame Ella Campbell Herbarium  |   |  |  | |
| LINC | co253 | Lincoln University Herbarium  |  |  |  | |
| UNITEC | co218 | Unitec Inst. of Tech. Herbarium |  |   |  | |

**Comment by rosemaryjoconnor** [comment]
** 22/07/2024 **

Checked for datasets in IPT Updated table above
Databox
- [ ] dr22717 - WELT loaded
- [ ] dr22783 - PDD loaded

Prod
- [ ] dr2153 - Still has whole original dataset with all records 
- [ ] dr26642 - WELT Images and data loaded

**Comment by rosemaryjoconnor** [comment]
**22/07/2024** - status update

dr2153 is made up of 9 collections. Only 3 of these are currently in IPT that I can find:

- Allan Herbarium (CHR) - dr2153
- WELT (WELT) - dr22717
- NZ Fungal and Plant Disease (PDD) - dr22783

**Databox**
All 3 of these have data in databox. However, dr2153 is not updated with only CHR records as that would mean the data for all the other collections would be missing from ALA
We need to find out the source of those, but I have no idea where to go for that. I've searched what I can in IPT and GBIF - but will try again. Provider maps are all fine.

**Prod**

- dr2153 - as per databox is not updated via IPT at this stage.
- dr26642 - WELT is loaded, waiting for Index to run overnight so that images are incorporated.

Peggy has said that for WELT we don't need to worry about the AVRO updates as neither dr2153 or dr26642 will be pushed to GBIF. This was clarified in support request that highlighted the duplication in GBIF. All that got sorted.

**What I would like to do is load:**

- NZ Fungal and Plan Disease (PDD) in prod via IPT - I can get metadata from GBIF
- Remove WELT data from dr2153 and reload (nice to have but doesn't have to be done yet) 

**What we need to find out:**

- The source of the other collections data so we can do the same (might be wishful thinking)
- Whether or not these are to be pushed to GBIF and so determine if we need to make AVRO updates (urgggh)
- Set up Provider maps for the outstanding datasets if they are to be loaded

**Comment by rosemaryjoconnor** [comment]
**13/08/2024**
Discussed with Mahmoud. No real idea what to do from here as there is no information re where the data for collections is coming from, apart from those available in IPT.

Slack message: https://atlaslivingaustralia.slack.com/archives/G0106GABXC3/p1716384230610869 
Slack messsage says:
So, DRs are: 
- AK: dr26650 (has not been uploaded yet, because behind firewall)
- CHR: dr26642 (this is the old NZVH dr) 
- PDD: dr26651
- WELT: dr26650

The above is not quite correct. What is actually in Prod is:

**Production**

- **AK:** dr26650 https://collections.ala.org.au/dataResource/show/dr26650 - No data
- **CHR**: dr2153 https://collections.ala.org.au/dataResource/show/dr2153 - original NZVH dataresource but has been named Allan Herbarium
- **PDD:** dr26651 https://collections.ala.org.au/dataResource/show/dr26651 No data
- **WELT:** dr26642 https://collections.ala.org.au/dataResource/show/dr26642 - 251,745 records

**Comment by rosemaryjoconnor** [comment]
**14/08/2024**

### Dataset status

As per tables above datasets for **AK, CHR, PDD, WELT** have been extracted via IPT:

- [ ] Successfully loaded in Databox and Production
- [ ] No authentication was needed for any
- [ ] Provider maps are working correctly
- [ ] Not shareable with GBIF flag set as requested per WELT support request 
- [ ] UUID avros have not been updated as datasets not shared with GBIF 

Data source for **NZFR, CANU, MPN, LINC,UNITEC** is yet to be determined.

### Dataresource dr2153 - NZ Herbarium

- [ ] Contains original full dataset for NZ Herbarium
- [ ] No shareable with GBIF set as requested per WELT support request

**Comment by rosemaryjoconnor** [comment]
**19/08/2024**

- Prod: Removed collections **['AK','CHR','PDD','WELT']**  from **dr2153**

**Record counts**

- Old record count:   1,005,699
- [ ]  New record count:  78,997 - waiting for index to run

**Comment by rosemaryjoconnor** [comment]
**27/08/2024**  Mtg NK and MS

- Work done to date is correct.
- Niels yet to receive datasets for remaining DRs

To Do:
- [ ] Rose to update Data Provider for AK','CHR','PDD','WELT'
- [ ] Metadata to be updated for AK','CHR','PDD','WELT'

**Note:** This issue to be closed and new issues raised when datasets for NZFR, CANU, MPN, LINC, UNITEC are available

**Comment by rosemaryjoconnor** [comment]
Hi Niels,
How does a dataset get added to AVH Hub? Is it something that will need to be done for each of the datasets?

thanks
Rose
________________________________

**Comment by nielsklazenga** [comment]
@rosemaryjoconnor , you do that here: https://collections.ala.org.au/dataHub/show/dh9. And yes, that needs to be done for all data resources that are part of AVH. I must have done the other ones already before I went on leave.

Reload of CHR data failed in pre-ingestion, btw.

**Comment by nielsklazenga** [comment]
Okay, this time it worked. We'll see tomorrow.

**Comment by rosemaryjoconnor** [comment]
**19/09/2024**

@nielsklazenga - Databox update - can you take a look and let me know if it is what you want/expect. 

I don't have metadata for the collections so the collectory DRs look very sparse. If you would like that populated in databox any suggestions where to find the info would be great. 

**Databox**
I've run the code to split out the collections below from dr2153 in databox. I have not deleted any records from dr2153 in databox, as I need them there for testing. If I had been deleting, the split would leave only NZFRI in dr2153 this can be changed of course. There are also 43 records with no collectionCode - file attached. All round the process is straightforward. 

|Code|DR| Name | 
|-----| ------|-------|
|NZFRI| [dr2153](https://collections-test.ala.org.au/dataResource/show/dr2153)  |National Forestry Herbarium | 
|CANU |[ dr22785](https://collections-test.ala.org.au/dataResource/show/dr22785) |  Univ of Canterbury Herbarium | 
|MPN| [dr22786](https://collections-test.ala.org.au/dataResource/show/dr22786) | Dame Ella Campbell Herbarium  |
|LINC|  [dr22787 ](https://collections-test.ala.org.au/dataResource/show/dr22787)|  Lincoln University Herbarium  | 
|UNITEC|  [dr22788](https://collections-test.ala.org.au/dataResource/show/dr22787) |  Unitec Inst. of Tech. Herbarium |  

**Unique identifiers and UUID Avro updates**

Based on discussions while you were away, UUID avro updates would only be needed if the source of the above are not IPT and would need to be pushed to GBIF. However I will finalise the code for it this week or early next week and thoroughly test the process in databox, just in case we need it. That will mean deleting one of the collections from dr2153 to test but that's ok. It's a little more complex than just changing the DR on the uniqueKey in the avro as I have to extract the correct avro records based on occurrenceIDs for each specific collection. I've done this previously for WELT, this just will be set up to work for all of the collections above in one go. 

I can incorporate some of Mahmoud's code to backup current Avros  on S3 and do the upload to S3 directly.

Attached: CSV file of records with no collectionCode: [nan-collectionCode.csv](https://github.com/user-attachments/files/17051565/nan-collectionCode.csv)

**Comment by nielsklazenga** [comment]
Hi @rosemaryjoconnor , there is no need to do anything with the UUIDs, as we do not deliver this data to GBIF and there is a good chance that when new datasets are delivered the `catalogNumber`s will have changed anyway. I was just going to set up a new DR when a collection provides us with a new dataset and remove that collection's data from dr2153 and eventually delete dr2153 (or leave it empty).

**Comment by rosemaryjoconnor** [comment]
@nielsklazenga  great Niels. I'll just leave it all as-is then. So no need to set up any DRs at this stage. If you do want any of the code just let me know, it will be in github anyway.

**Comment by rosemaryjoconnor** [comment]
**24/09/2024**

**Databox**
New data for NZFRI is available:
- dr2153 - removed 'NZFRI', 'AK', 'CHR', 'PDD', 'WELT' collections22784
- dr22784 - Niels testing

**Comment by rosemaryjoconnor** [comment]
**26/09/2024**

**Prod**

- dr2153 - successfully updated - index successful. 
- New record count: 50,063
